src/components/Events.tsx

```diff
@@ -1,44 +1,44 @@
-export const mockEvents = [
-  {
-    id: 1,
-    name: "Vendy",
-    sport: "Basketball",
-  },
-  {
-    id: 2,
-    name: "Lukas",
-    sport: "Football",
-  },
-  {
-    id: 3,
-    name: "Honza",
-    sport: "Tennis",
-  },
-  {
-    id: 4,
-    name: "Vendy",
-    sport: "Box",
-  },
-  {
-    id: 5,
-    name: "Lukas",
-    sport: "Skydiving",
-  },
-  {
-    id: 6,
-    name: "Johana",
-    sport: "Jogging",
-  },
-];
+import { useState, useEffect } from "react";
 
-export const Events: React.FC = () => (
-  <>
-    {mockEvents.map(({ id, sport, name }) => (
-      <article key={id}>
-        <p>{sport}</p>
-        <p>od: {name}</p>
-        <button>Otevřít</button>
-      </article>
-    ))}
-  </>
-);
+interface Event {
+  id: number;
+  sport: string;
+  name: string;
+}
+
+const fetchEventsData = async () => {
+  const response = await fetch("https://sportujspolu-api.onrender.com/events");
+  if (!response.ok) {
+    throw new Error("Failed to fetch data");
+  }
+  const events = await response.json();
+  return events as Event[];
+};
+
+export const Events: React.FC = () => {
+  const [events, setEvents] = useState<Event[]>([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const events = await fetchEventsData();
+        setEvents(events);
+        console.log("events:", events);
+      } catch (error) {
+        console.error(error);
+      }
+    })();
+  }, []);
```

Comment:
So this is called IIFE - immediately invoked function expression. Basically, a function that calls itself. Definitely, something you could get a question on an interview :D.

But in real code, there are close to 0 reasons this needs to be used. It is hard to read, and very messy and juniors won't know it. I know you have a lot on UD, but it is quite an anti-pattern.

A good pattern and maybe even good practice, in this case, would be this: 

```suggestion
  useEffect(() => {
    const getData = async () => {
      try {
        const events = await fetchEventsData();
        setEvents(events);
        console.log("events:", events);
      } catch (error) {
        console.error(error);
      }
    }
    
    getData();
  }, []);
```

Reply:
Oh I see, that was my first bet! But then I recalled this thing - same result and functionally is equivalent. Also,  I like that it is weird and concise. But I do not have an eye for good patterns and best practices yet!  (And I am messy, heh)